src/request/arrive_before.rs

```diff
@@ -259,6 +259,25 @@ where
         Some(new_criteria)
     }
 
+    fn stay_in(&self, trip: &Data::Trip, criteria: &Criteria) -> Option<(Data::Trip, Criteria)> {
+        let previous_trip = self
+            .transit_data
+            .stay_in_previous(trip, self.real_time_level)?;
+        let mission = self.transit_data.mission_of(&previous_trip);
+        let last_position = self.transit_data.last_on_mission(&mission);
+        let (arrival_time_at_last_stop, load) = self
+            .transit_data
+            .arrival_time_of(&previous_trip, &last_position);
+        let new_criteria = Criteria {
+            time: arrival_time_at_last_stop,
```

Comment:
```suggestion
            time: departure_time_at_last_stop,
```
to be coherent with https://github.com/CanalTP/loki/blob/d24f0028dc243f2c013a6b389e6c70af1cd1de1e/src/request/arrive_before.rs#L251